src/utils.ts

```diff
@@ -52,23 +67,119 @@ export function fixMantissaOverflow(
+ */
+export function toLongScale(value: Decimal): Decimal {
+  // Give numbers between a thousand and a million exponent 3.
+  const mod = value.exponent < 6 ? 3 : 6;
+  const exponentOffset = value.exponent % mod;
+  return Decimal.fromMantissaExponent_noNormalize(
+    value.mantissa * 10 ** exponentOffset,
     value.exponent - exponentOffset
   );
 }
 
 export function toFixedEngineering(value: Decimal, places: number): Decimal {
-  return fixMantissaOverflow(toEngineering(value), places, 1000, 3);
+  return fixMantissaOverflow(
+    toEngineering(value),
+    places,
+    1000,
+    3
+  );
+}
+
+export function toFixedLongScale(value: Decimal, places: number): Decimal {
+  const overflowPlaces = value.exponent < 6 ? 3 : 6;
+  return fixMantissaOverflow(
+    toLongScale(value),
+    places,
+    10 ** overflowPlaces,
+    overflowPlaces
+  );
 }
 
 const SUBSCRIPT_NUMBERS = ["₀", "₁", "₂", "₃", "₄", "₅", "₆", "₇", "₈", "₉"];
 
 export function toSubscript(value: number): string {
-  return value.toFixed(0).split("").map(x => SUBSCRIPT_NUMBERS[parseInt(x)]).join("");
+  return value.toFixed(0).split("")
+    .map((x) => x === '-' ? '₋' : SUBSCRIPT_NUMBERS[parseInt(x)])
+    .join("");
 }
 
 const SUPERSCRIPT_NUMBERS = ["⁰", "¹", "²", "³", "⁴", "⁵", "⁶", "⁷", "⁸", "⁹"];
 
 export function toSuperscript(value: number): string {
-  return value.toFixed(0).split("").map(x => SUPERSCRIPT_NUMBERS[parseInt(x)]).join("");
+  return value.toFixed(0).split("")
+    .map((x) => x === '-' ? '⁻' : SUPERSCRIPT_NUMBERS[parseInt(x)])
+    .join("");
+}
+
+const STANDARD_ABBREVIATIONS = [
+  "K", "M", "B", "T", "Qa", "Qt", "Sx", "Sp", "Oc", "No", "Dc", "UDc", "DDc",
+  "TDc", "QaDc", "QtDc", "SxDc", "SpDc", "ODc", "NDc", "Vg", "UVg", "DVg", "TVg",
+  "QaVg", "QtVg", "SxVg", "SpVg", "OVg", "NVg", "Tg", "UTg", "DTg", "TTg", "QaTg",
+  "QtTg", "SxTg", "SpTg", "OTg", "NTg", "Qd", "UQd", "DQd", "TQd", "QaQd", "QtQd",
+  "SxQd", "SpQd", "OQd", "NQd", "Qi", "UQi", "DQi", "TQi", "QaQi", "QtQi", "SxQi",
+  "SpQi", "OQi", "NQi", "Se", "USe", "DSe", "TSe", "QaSe", "QtSe", "SxSe", "SpSe",
+  "OSe", "NSe", "St", "USt", "DSt", "TSt", "QaSt", "QtSt", "SxSt", "SpSt", "OSt",
+  "NSt", "Og", "UOg", "DOg", "TOg", "QaOg", "QtOg", "SxOg", "SpOg", "OOg", "NOg",
+  "Nn", "UNn", "DNn", "TNn", "QaNn", "QtNn", "SxNn", "SpNn", "ONn", "NNn", "Ce"
+];
+
+const STANDARD_PREFIXES = [
+  ["", "U", "D", "T", "Qa", "Qt", "Sx", "Sp", "O", "N"],
+  ["", "Dc", "Vg", "Tg", "Qd", "Qi", "Se", "St", "Og", "Nn"],
+  ["", "Ce", "Dn", "Tc", "Qe", "Qu", "Sc", "Si", "Oe", "Ne"]
+];
+
+const STANDARD_PREFIXES_2 = ["", "MI-", "MC-", "NA-", "PC-", "FM-"];
+
+export function abbreviate(e: number): string {
+  // Please, someone clean this code up eventually
+  if (e < STANDARD_ABBREVIATIONS.length) {
+    return STANDARD_ABBREVIATIONS[e];
+  }
+  let index2 = 0;
+  const prefix = [STANDARD_PREFIXES[0][e % 10]];
+  while (e >= 10) {
+    e = Math.floor(e / 10);
```

Comment:
Codacy found an issue: [Reassigning parameter 'e' is forbidden.](https://app.codacy.com/gh/antimatter-dimensions/notations/pullRequest?prid=6218257)